internal/services/firewall/firewall_resource.go

```diff
@@ -224,6 +224,31 @@ func resourceFirewall() *pluginsdk.Resource {
 				},
 			},
 
+			"autoscale_configuration": {
+				Type:        pluginsdk.TypeSet,
```

Comment:
Please change this to `pluginsdk.TypeList` as the type set has issues in the legacy SDK. Also, please add an additional field: `MaxItems: 1` to indicate that this is a nested object.